sdk/core/core-rest-pipeline/src/interfaces.ts

```diff
@@ -47,6 +47,39 @@ export interface HttpHeaders extends Iterable<[string, string]> {
   toJSON(options?: { preserveCase?: boolean }): RawHttpHeaders;
 }
 
+/**
+ * A part of the request body in a multipart request.
+ */
+export interface BodyPart {
+  /**
+   * The headers for this part of the multipart request.
+   */
+  headers: HttpHeaders;
+
+  /**
+   * The body of this multipart request.
+   */
+  body: ReadableStream | NodeJS.ReadableStream | Uint8Array | Blob;
```

Comment:
I think we need to add `File` to the union because a multipart request could be a list of files or just contain a file or more.

Reply:
The doc string is misleading, this body only corresponds to one part of the multipart request. I will fix that.

Adding `File` here is unnecessary as `File` already satisfies `Blob` and none of the extra attributes `File` has are used by `multipartPolicy`.

Reply:
I copied this example from [SO](https://stackoverflow.com/a/66551704/499148) and I see filename being inlined, shouldn't we be able to create something like it?:
```
From: "no-reply @ Example" no-reply@example.com
To: no-reply@example.com
Subject: Important information
Message-Id: 0eb69c12-ec18-4f6c-ac14-168b2092b477@example.com
Content-Language: en-US
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="mixed_boundary"

--mixed_boundary
Content-Type: multipart/related; boundary="related_boundary"

--related_boundary
Content-Type: multipart/alternative; boundary="alternative_boundary"

--alternative_boundary
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 8bit

Dear recipient,

Please find attached three files with important information.

You may also view the three files bellow:

[cid:50eef9fb]
[cid:e338d1eb]
[cid:e98e57c2]

MAX-76-CHAR-MAX-76-CHAR-MAX-76-CHAR-MAX-76-CHAR-MAX-76-CHAR-MAX-76-CHAR-MAX-

(c) 2023 - Example, Inc.

Message Id: 0eb69c12-ec18-4f6c-ac14-168b2092b477

--alternative_boundary
Content-Type: text/html; charset="utf-8"
Content-Transfer-Encoding: 8bit

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html>
<body style="font-family: consolas, monospace;">
<p>
Dear recipient,
</p>
<p>
Please find attached three files with important information.
</p>
<p>
You may also view the three files bellow:
</p>
<ol>
<li><img src="cid:50eef9fb" alt="Red"></li>
<li><img src="cid:e338d1eb" alt="Green"></li>
<li><img src="cid:e98e57c2" alt="Blue"></li>
</ol>
<p>
MAX-76-CHAR-MAX-76-CHAR-MAX-76-CHAR-MAX-76-CHAR-MAX-76-CHAR-MAX-76-CHAR-MAX-
</p>
<p>
(c) 2023 - Example, Inc.
</p>
<p>
Message Id: 0eb69c12-ec18-4f6c-ac14-168b2092b477
</p>
</body>
</html>

--alternative_boundary--

--related_boundary
Content-Type: image/gif; name="red"
Content-Description: red.gif
Content-Disposition: inline; filename="image001.gif";
Content-ID: <50eef9fb>
Content-Transfer-Encoding: base64

R0lGODdhIAAgAPAAAPRDNslFJiH5BAEAAAEALAAAAAAgACAAAAIehI+py+0Po5y02ouz3rz7D4bi
SJbmiabqyrbuC5sFADs=

--related_boundary
Content-Type: image/gif; name="green"
Content-Description: green.gif
Content-Disposition: inline; filename="image002.gif";
Content-ID: <e338d1eb>
Content-Transfer-Encoding: base64

R0lGODdhIAAgAPAAAC59MslFJiH5BAEAAAEALAAAAAAgACAAAAIehI+py+0Po5y02ouz3rz7D4bi
SJbmiabqyrbuC5sFADs=

--related_boundary
Content-Type: image/gif; name="blue"
Content-Description: blue.gif
Content-Disposition: inline; filename="image003.gif";
Content-ID: <e98e57c2>
Content-Transfer-Encoding: base64

R0lGODdhIAAgAPAAABVlwMlFJiH5BAEAAAEALAAAAAAgACAAAAIehI+py+0Po5y02ouz3rz7D4bi
SJbmiabqyrbuC5sFADs=

--related_boundary--

--mixed_boundary
Content-Type: image/gif; name="red"
Content-Description: red.gif
Content-Disposition: attachment; filename="image004.gif";
Content-Transfer-Encoding: base64

R0lGODdhIAAgAPAAAPRDNslFJiH5BAEAAAEALAAAAAAgACAAAAIehI+py+0Po5y02ouz3rz7D4bi
SJbmiabqyrbuC5sFADs=

--mixed_boundary
Content-Type: image/gif; name="green"
Content-Description: green.gif
Content-Disposition: attachment; filename="image005.gif";
Content-Transfer-Encoding: base64

R0lGODdhIAAgAPAAAC59MslFJiH5BAEAAAEALAAAAAAgACAAAAIehI+py+0Po5y02ouz3rz7D4bi
SJbmiabqyrbuC5sFADs=

--mixed_boundary
Content-Type: image/gif; name="blue"
Content-Description: blue.gif
Content-Disposition: attachment; filename="image006.gif";
Content-Transfer-Encoding: base64

R0lGODdhIAAgAPAAABVlwMlFJiH5BAEAAAEALAAAAAAgACAAAAIehI+py+0Po5y02ouz3rz7D4bi
SJbmiabqyrbuC5sFADs=

--mixed_boundary--
```

Reply:
you meant the `Content-Description` header? I remember that [the spec dis-allows](https://www.rfc-editor.org/rfc/rfc7578#section-4.8) other `Content-*` headers, so this may not be spec-compliant.

Reply:
Thanks for clarifying, I am fine then with what we have but perhaps we should log an issue for supporting nested bodies.

Also, do you think the property name should be renamed to something clearer? `bodyPart` or `part` or something?

Reply:
In practice declaring a body will look like this:

```ts
request.multipartBody = {
  boundary: /* ... */,
  parts: [
    {
      headers: /* ... */,
      body: /* ... */,
    },
    {
      headers: /* ... */,
      body: /* ... */,
    },
    // ...
  ]
};
```

IMO the extra context of `parts:` in MultipartRequestBody provides the necessary clarity, is that OK with you?

Reply:
Sounds good to me, thanks!

Reply:
I also realized I didn't answer @jeremymeng's question, I meant the `Content-Disposition` one.